### PR TITLE
Minimal cleanup of PFClusterValidation code

### PR DESCRIPTION
**Pull request by fwyzard**
Minimal clean up of the PFClusterValidation module, addressing various issues:
  - unsafe assumptions that could potentially "leak" information across events;
  - use of common functions instead of private redefinitions;
  - adherence to CMS coding rules.

### CODE REVIEW

**Comment by fwyzard** [comment]
Fixes #30588 .

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
@cmsbuild, please test

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 1]
```diff
@@ -1,271 +1,233 @@
...
-  }
 }
 
 PFClusterValidation::~PFClusterValidation() {}
 
 void PFClusterValidation::bookHistograms(DQMStore::IBooker& ibooker,
                                          edm::Run const& irun,
                                          edm::EventSetup const& isetup) {
-  Char_t histo[100];
+  static const auto size = 100;
```
Comment (Dr15Jones) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:26]:
```suggestion
  constexpr auto size = 100;
```

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 2]
```diff
@@ -1,271 +1,233 @@
...
+  char histo[size];
   ibooker.setCurrentFolder("ParticleFlow/PFClusterV");
-  Double_t etaBinsOffset[83] = {
+  double etaBinsOffset[83] = {
```
Comment (Dr15Jones) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:29]:
```suggestion
  constexpr double etaBinsOffset[83] = {
```

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 3]
```diff
@@ -1,271 +1,233 @@
...
-  double HFcone = 0.;
-  double HOcone = 0.;
-
-  // sum the energy in a dr cone for each subsystem
-  Econe = sumEnergy(pfClusterECAL);
-  Hcone = sumEnergy(pfClusterHCAL);
-  HOcone = sumEnergy(pfClusterHO);
-  HFcone = sumEnergy(pfClusterHF);
+  // sum the energy in a dR cone for each subsystem
+  double Econe = sumEnergy(pfClusterECAL, eta_MC, phi_MC);
```
Comment (Dr15Jones) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:165]:
```suggestion
  const double Econe = sumEnergy(pfClusterECAL, eta_MC, phi_MC);
```

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 4]
```diff
@@ -1,271 +1,233 @@
...
+  double Hcone = sumEnergy(pfClusterHCAL, eta_MC, phi_MC);
```
Comment (Dr15Jones) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:166]:
```suggestion
  const double Hcone = sumEnergy(pfClusterHCAL, eta_MC, phi_MC);
```

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 5]
```diff
@@ -1,271 +1,233 @@
...
+  double HOcone = sumEnergy(pfClusterHO, eta_MC, phi_MC);
```
Comment (Dr15Jones) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:167]:
```suggestion
  const double HOcone = sumEnergy(pfClusterHO, eta_MC, phi_MC);
```

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc [hunk 6]
```diff
@@ -1,271 +1,233 @@
...
+  double HFcone = sumEnergy(pfClusterHF, eta_MC, phi_MC);
```
Comment (Dr15Jones) [inline Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:168]:
```suggestion
  const double HFcone = sumEnergy(pfClusterHF, eta_MC, phi_MC);
```

**Review by Dr15Jones — COMMENTED** [review]
Thanks for making these changes!

File: Validation/RecoParticleFlow/plugins/PFClusterValidation.cc (hunk 3 above)
Comment (fwyzard) [reply Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:165]: do these make any difference? the compiler can see that they are not being changed.

Comment (Dr15Jones) [reply Validation/RecoParticleFlow/plugins/PFClusterValidation.cc:165]: It makes it clear to future developers that the values are not supposed to be changed

**Comment by fwyzard** [comment]
Thanks for the comments @Dr15Jones , they should all be addressed in the last commit.

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**